File: javascript_to_golang/experiment7.go
package main

import "fmt"

func logPairs(array1 []string, array2 []string, words string) {
	counter := 0
	for i := 0; i < len(array1); i++ {
		for j := 0; j < len(array2); j++ {
			counter++
			fmt.Println(words, counter, array1[i], "dan", array2[j])
		}
	}
}

func main() {
	foods := []string{"Sate", "Bakso", "Dimsum", "Rames"}
	drinks := []string{"Jeruk", "Teh", "Kelapa", "Cendol"}
	logPairs(foods, drinks, "Menu")
}

File: javascript_to_java/experiment-7.java
class experiment7 {
    public static void main(String[] args) {
        String[] foods = { "Sate", "Bakso", "Dimsum", "Rames" };
        String[] drinks = { "Jeruk", "Teh", "Kelapa", "Cendol" };
        logPairs(foods, drinks, "Menu");
    }

    public static void logPairs(String[] array1, String[] array2, String words) {
        int counter = 0;
        for (int i = 0; i < array1.length; i++) {
            for (int j = 0; j < array2.length; j++) {
                counter++;
                System.out.println(words + ' ' + counter + ' ' + array1[i] + " dan " + array2[j]);
            }
        }
    }
}